pysalesforceutils.py

```diff
@@ -712,6 +713,105 @@ def get_sobject_row(object_name, record_id, field_list_string, access_token, ins
+            response = Standard.get_sobject_blob('Attachment', 'attachmentId', access_token, instance_url)
+            with open('test.jpg', 'wb') as new_file:
+                for chunk in response.iter_content(chunk_size=1024):
+                    new_file.write(chunk)
+
+        Args:
+            object_name (str): The API name of the object.
+            record_id (str): The record Id of the Attachment you're trying to retrieve
+            access_token (str): This is the access_token value received from the
+                                login response
+            instance_url (str): This is the instance_url value received from the
+                                login response
+
+        Returns:
+            object: returns the record with the explicit field list, or all (or
+                    a lot) of the fields if the field_list_string is None.
+        """
+        object_to_blob_map = {
+            "Attachment": "Body",
+            "ContentVersion": "VersionData",
+            "Document": "Body"
+        }
+
+        get_row_uri = '/sobjects/' + object_name + '/' + record_id + '/' + object_to_blob_map[object_name]
+        header_details = Util.get_standard_header(access_token)
+
+        response = webservice.Tools.get_http_response(
+            instance_url + Standard.base_standard_uri + 'v' + API_VERSION + get_row_uri, header_details)
+
+        return response
+
+    @staticmethod
+    def create_sobject_blob_record(object_name, record_json, file, access_token, instance_url):
+        """
+        Creates the provided record in the recordJson param and binary multipart message
+
+        Args:
+            object_name (str): The API name of the object.
+            record_json (object): The JSON describing the fields you want to
+                                  update on the given object. You should pass in
+                                  a python object and it will be converted to a
+                                  json string to send the request. This object
+                                  is just the key value paris for the record
+                                  update. e.g.:
+                                      {
+                                          'BillingCity': 'Bellevue',
+                                          'BillingState': 'WA'
+                                      }
+            file (str): file object
+            access_token (str): This is the access_token value received from the
+                                login response
+            instance_url (str): This is the instance_url value received from the
+                                login response
+        Returns:
+            object: returns the text from the creation response
+        """
+        post_row_uri = '/sobjects/'+ object_name + '/'
+        header_details = {"Authorization": "Bearer " + access_token}
+        mimetype = MimeTypes().guess_type(os.path.basename(file.name))[0] or 'application/octet-stream'
```

Comment:
I wasn't sure if os.path.basename was needed but decided to strip it incase it was causing issues with the mimetype guessing.